File: v6/prolog/compile/test/typegen_golden/shape_option_nested_enum.types.ts
export type Option<T> = { tag: 'none' } | { tag: 'some'; value: T };

export type Status =
  | { tag: 'ready' }
  | { tag: 'failed'; reason: string; }
;

export interface StatusReady {
  id: number;
}

export interface StatusFailed {
  id: number;
  reason: string;
}

export interface Job {
  id: number;
  state: Option<Status>;
  nested_state: Option<Option<Status>>;
}

File: v6/prolog/compile/test/typegen_golden/shape_option_nested_enum.types.rs
#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
#[serde(tag = "tag", content = "value", rename_all = "snake_case")]
pub enum DlOption<T> {
    None,
    Some(T),
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub enum Status {
    Ready,
    Failed { reason: String },
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct StatusReady {
    pub id: i64,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct StatusFailed {
    pub id: i64,
    pub reason: String,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct Job {
    pub id: i64,
    pub state: DlOption<Status>,
    pub nested_state: DlOption<DlOption<Status>>,
}
